tasks/libs/github_actions_tools.py

```diff
@@ -86,20 +90,35 @@ def trigger_macos_workflow(
     # Thus the following hack: Send an id as input when creating a workflow on Github. The worklow will use the id and put it in the name of one of its jobs.
     # We then fetch workflows and check if it contains the id in its job name.
 
+    # Adding another hack to check if the workflow is not waiting a concurrency to be solved before generating the workflow jobs
+    might_be_waiting = set()
+
+    MAX_WAITING_CONCURRENCY_RETRIES = 3  # Retries for up to 1h30
     MAX_RETRIES = 30  # Retry for up to 5 minutes
-    for i in range(MAX_RETRIES):
-        print(f"Fetching triggered workflow (try {i + 1}/{MAX_RETRIES})")
-        recent_runs = gh.workflow_run_for_ref_after_date(workflow_name, github_action_ref, now)
-        for recent_run in recent_runs:
-            jobs = recent_run.jobs()
-            if jobs.totalCount >= 2:
-                for job in jobs:
-                    if any([step.name == workflow_id for step in job.steps]):
-                        return recent_run
-            else:
-                print("waiting for jobs to popup...")
-                sleep(5)
-        sleep(10)
+    for i in range(MAX_WAITING_CONCURRENCY_RETRIES):
+        for j in range(MAX_RETRIES):
+            print(f"Fetching triggered workflow (try {j + 1}/{MAX_RETRIES})")
+            recent_runs = gh.workflow_run_for_ref_after_date(workflow_name, github_action_ref, now)
+            for recent_run in recent_runs:
+                jobs = recent_run.jobs()
+                if jobs.totalCount == 0:
+                    might_be_waiting.add(recent_run.id)
+                if jobs.totalCount >= 2:
```

Comment:
I'm not sure but should this be `elif` instead ? Otherwise the `else` might not be executed
```suggestion
                elif jobs.totalCount >= 2:
```

Reply:
This is a different block so I don't want an `elif`.
However to make it better I can update the `might_be_waiting` in the `else`, yes you are right, thanks